test/transaction/payload/proregtxpayload.js

```diff
@@ -0,0 +1,148 @@
+var expect = require('chai').expect;
+
+var DashcoreLib = require('../../..');
+
+var BufferUtil = DashcoreLib.util.buffer;
+var Payload = DashcoreLib.Transaction.Payload;
+var ProRegTxPayload = Payload.ProRegTxPayload;
+
+var options = {
+  type: 1,
+  mode: 2,
+  collateralIndex: 1,
+  ipAddress: '0000:0000:0000:0000:0000:ffff:c0a8:0001', //192.168.0.1
+  port: 10000,
+  KeyIdOwner: '4d5fce2325deb034ae75a625a3e2f09395e27bf7',
+  KeyIdOperator: '4d5fce2325deb034ae75a625a3e2f09395e27bf7',
+  KeyIdVoting: '4d5fce2325deb034ae75a625a3e2f09395e27bf7',
+  operatorReward: 100,
+  scriptPayout: 'deb034ae',
+  inputsHash: 'a6f7b4284fb753eab9b554283c4fe1f1d7e143e6cf3b975d0376d7c08ba4cdf5',
+  payloadSig: '48d6a1bd2cd9eec54eb866fc71209418a950402b5d7e52363bfb75c98e141175',
+}
+
+
+describe('ProRegTxPayload', function () {
+
+  describe('constructor', function () {
+    it('Should create ProRegTxPayload instance', function () {
+      var payload = new ProRegTxPayload();
+      expect(payload).to.have.property('version');
+    });
+  });
+
+  describe('from-/to- Buffer', function () {
+    it('Should return instance of ProRegTxPayload with parsed data', function () {
+
+      var payloadBuffer = new ProRegTxPayload(options).toBuffer();
+      expect(BufferUtil.isBuffer(payloadBuffer)).to.be.true;
+
+      var parsedPayload = ProRegTxPayload.fromBuffer(payloadBuffer);
+      for (var key in options) {
+        expect(parsedPayload[key]).to.be.equal(options[key]);
+      }
+    });
+  });
+
+  describe('from- / to- JSON', function () {
+    it('Should parse JSON', function () {
+
+      options.version = 1;
+      var payloadObj = new ProRegTxPayload().fromJSON(options);
```

Comment:
`.fromJSON` should be a static method. It's used as static when parsing special transaction: https://github.com/dashevo/dashcore-lib/blob/evo/lib/transaction/payload/payload.js#L43

It also worth noting that payload class needs to be included here: https://github.com/dashevo/dashcore-lib/blob/evo/lib/transaction/payload/payload.js#L10

Reply:
updated